### PR TITLE
Search form

### PR DESCRIPTION
**Pull request by ocskier**
Have the distance and checkboxes filtering data

### CODE REVIEW

**Comment by ocskier** [comment]
Working on #28

File: client/src/components/Form/Form.js [hunk 1]
```diff
@@ -1,43 +1,74 @@
...
+import FormControl from '@material-ui/core/FormControl';
+import Select from '@material-ui/core/Select';
 
 // Local imports
-import DistanceSlider from "./DistanceSlider";
-import CheckboxGroup from "./CheckboxGroup";
+import DistanceSlider from './DistanceSlider';
+import CheckboxGroup from './CheckboxGroup';
+
+import data from '../../data/demo.json';
```
Comment (tlburchh) [inline client/src/components/Form/Form.js:16]: scripts/seedDB.js has our most up to date seed data.

Comment (tlburchh) [reply client/src/components/Form/Form.js:16]: Better to do an api call?

Comment (krab7191) [reply client/src/components/Form/Form.js:16]: Yeah remove this, implement API call to get seeded data.

File: client/src/components/Form/Form.js [hunk 2]
```diff
@@ -1,43 +1,74 @@
...
+
+function haversineCalc(la1,la2,lo1,lo2){
```
Comment (krab7191) [inline client/src/components/Form/Form.js:18]: We already have geolocation baked into the map, lets either tap into that or replace it so we don't have 2 implementations that could get out of sync and harder to maintain.

File: client/src/components/Form/Form.js [hunk 3]
```diff
@@ -1,43 +1,74 @@
...
+      display: 'inline-grid',
     },
     clearBtn: {
-      fontSize: "0.7rem",
+      fontSize: '0.7rem',
     },
   })
 );
 
-export default function Form({ setResults }) {
```
Comment (krab7191) [inline client/src/components/Form/Form.js:37]: I renamed this in my PR, but lets be more descriptive with component names, we don't want to have to look at code to determine what kind of form it is.

File: client/src/components/Form/Form.js [hunk 4]
```diff
@@ -67,49 +98,81 @@ export default function Form({ setResults }) {
...
             value={source}
             onChange={handleChangeSource}
             displayEmpty
             className={classes.selectEmpty}
           >
-            <MenuItem value="All">
+            <MenuItem value='All'>
               <em>All</em>
             </MenuItem>
-            <MenuItem value={"Farm"}>Farm</MenuItem>
```
Comment (krab7191) [inline client/src/components/Form/Form.js:88]: Can easily be mapped to keep code DRY

File: client/src/components/leafletMap/leafletMap.js [hunk 5]
```diff
@@ -43,24 +43,26 @@ export const LeafletMap = () => {
...
-          position={[
-            biz.location[0].lat,
-            biz.location[0].lng
-          ]}
-          onClick={() => {
-            setActiveBusiness(biz);
-          }}
-        />
-      ))}
+      {results &&
```
Comment (krab7191) [inline client/src/components/leafletMap/leafletMap.js:46]: Again here lets keep names more descriptive, I don't know what results are, but businessData gives me an idea.

File: client/src/pages/home/home.js [hunk 6]
```diff
@@ -1,40 +1,50 @@
...
-import Grid from "@material-ui/core/Grid";
-import Form from "../../components/Form";
-import LMap from "../../components/leafletMap";
-import Results from "../../components/Results";
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+
+import { withStyles } from '@material-ui/core/styles';
+import Paper from '@material-ui/core/Paper';
+import Grid from '@material-ui/core/Grid';
```
Comment (krab7191) [inline client/src/pages/home/home.js:6]: Can be seen in my PR, lets consolidate our MUI core imports into 1 line.

File: client/src/pages/home/home.js [hunk 7]
```diff
@@ -45,12 +55,12 @@ function Home(props) {
         <Grid item xs={12}>
           <Grid item xs={4}></Grid>
           <Paper elevation={0} className={classes.map}>
-            <LMap />
```
Comment (krab7191) [inline client/src/pages/home/home.js:48]: Can see in my PR, LeafletMap to be more descriptive.

File: client/src/components/Form/Form.js (hunk 2 above)
Comment (ocskier) [reply client/src/components/Form/Form.js:18]: i can tap into the map geolocation perhaps for the current users location but the haversineCalc is there to calculate the filtered list to render to the map

Comment (krab7191) [reply client/src/components/Form/Form.js:18]: Yep makes sense, I just don't want to have 2 implementations of geolocation. Lets break the Haversine out into a separate file then, separation of concerns and what not

Comment (ocskier) [reply client/src/components/Form/Form.js:18]: Maybe move it to utils?

Comment (ocskier) [reply client/src/components/Form/Form.js:18]: Or just into a sep file in the Form dir?

Comment (krab7191) [reply client/src/components/Form/Form.js:18]: I'd say utils makes more sense.

**Pull request merged**